llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -8020,6 +8020,10 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
       if (Instruction *NV = foldOpIntoPhi(I, cast<PHINode>(LHSI)))
         return NV;
       break;
+    case Instruction::Select:
+      if (Instruction *NV = FoldOpIntoSelect(I, cast<SelectInst>(LHSI)))
+        return NV;
+      break;
```

Comment:
If both arms are constant, should we allow multiuse select?

Reply:
Yeah, we can relax it in `InstCombinerImpl::FoldOpIntoSelect`.

Reply:
IMO here, we have the argument `FoldWithMultiUse` and it would be a bit confusing if we started to surreptitiously assume it.